src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -547,41 +547,42 @@ public unsafe void WriteReferences(CurrentIndexingScope indexingScope, RavenTran
                 }
             }
 
-            if (indexingScope.ReferenceEtagsByCollection != null)
-            {
-                foreach (var kvp in indexingScope.ReferenceEtagsByCollection)
-                {
-                    var lastIndexedEtag = ReadLastIndexedEtag(tx, kvp.Key);
```

Comment:
Shouldn't we read here following:
```
var lastEtagInCollection = ...
var isLastEtagInCollection = lastIndexedEtag  == lastEtagInCollection;
```

and in `if (etag > lastIndexedEtag)` change to:

```
if (etag > lastIndexedEtag && isLastEtagInCollection == false)
```

?